pages/health.py

```diff
@@ -18,18 +33,548 @@
+
+def outcome_narrative(outcome_df, expenditure_df, country):
+    try:
+        start_year = expenditure_df.year.min()
+        end_year = expenditure_df.year.max()
+
+        merged = pd.merge(outcome_df, expenditure_df, on=["year"], how="inner")
+        x_col = {"display": "universal health coverage index", "col_name": "universal_health_coverage_index"}
+        y_col = {"display": "per capita public spending", "col_name": "per_capita_real_expenditure"}
+        PCC = get_correlation_text(merged, x_col, y_col)
+
+        text = f"From {start_year} to {end_year}, {PCC}"
+    except:
+        traceback.print_exc()
+        return generate_error_prompt("GENERIC_ERROR")
+    return text
+
+
+@callback(
+    Output("health-outcome", "figure"),
+    Output("health-outcome-measure", "children"),
+    Output("health-outcome-narrative", "children"),
+    Input("stored-data-health-outcome", "data"),
+    Input("stored-data-health-total", "data"),
+    Input("country-select", "value"),
 )
+def render_health_outcome(outcome_data, total_data, country):
+    if not total_data or not outcome_data:
+        return
+
+    uhc = pd.DataFrame(outcome_data["uhc_index"])
+    uhc = filter_country_sort_year(uhc, country)
+
+    pub_exp = pd.DataFrame(total_data["health_public_expenditure"])
+    pub_exp = filter_country_sort_year(pub_exp, country)
+
+    fig = make_subplots(specs=[[{"secondary_y": True}]])
+
+    fig.add_trace(
+        go.Scatter(
+            name="Universal health coverage index",
+            x=uhc.year,
+            y=uhc.universal_health_coverage_index,
+            mode="lines+markers",
+            line=dict(color="deeppink", shape="spline", dash="dot"),
+            connectgaps=True,
+        ),
+        secondary_y=True,
+    )
+
+    fig.add_trace(
+        go.Scatter(
+            name="Inflation adjusted per capita public spending",
+            x=pub_exp.year,
+            y=pub_exp.per_capita_real_expenditure,
+            mode="lines",
+            marker_color="darkblue",
+            opacity=0.6,
+        ),
+        secondary_y=False,
+    )
+
+    fig.update_layout(
+        plot_bgcolor="white",
+        height=500,
+        legend=dict(
+            orientation="h",
+            yanchor="bottom",
+            y=0.95,
+            xanchor="left",
+            x=0,
+            bgcolor="rgba(0,0,0,0)",
+        ),
+        title=dict(
+            text="How has health outcome changed?",
+            y=0.95,
+            x=0.5,
+            xanchor="center",
+            yanchor="top"
+        ),
+        annotations=[
+            dict(
+                xref="paper",
+                yref="paper",
+                x=-0,
+                y=-0.2,
+                text="Source: UHC: WHO; BOOST: World Bank; Population: UN, Eurostat",
+                showarrow=False,
+                font=dict(size=12),
+            )
+        ],
+    )
+
```

Comment:
We could decide not to cut off the hover text by adding
` fig.update_layout(hoverlabel_namelength=-1)
`
![image](https://github.com/user-attachments/assets/180f86cf-e9e8-42db-afda-7dcd1a2c46a0)

Reply:
Good catch! I've applied it to both education and health outcome plots